tests/unit/pip_sync_faster/main_test.py

```diff
@@ -1,6 +1,181 @@
+            """Cache the given hashes for the given paths."""
+            with open(cached_hashes_path, "w", encoding="utf-8") as handle:
+                json.dump(hashes, handle)
+
+        return cache_hashes
+
+    @pytest.fixture
+    def requirements_files(self, tmp_path):
+        """Metadata about the test requirements files."""
+        dev_txt_path = tmp_path / "dev.txt"
+        with open(dev_txt_path, "w", encoding="utf8") as handle:
+            handle.write("This is a fake dev.txt requirements file.")
+
+        format_txt_path = tmp_path / "format.txt"
+        with open(format_txt_path, "w", encoding="utf8") as handle:
+            handle.write("This is a fake format.txt requirements file.")
+
+        return Files([File(dev_txt_path), File(format_txt_path)])
+
+    @pytest.fixture
+    def test_it_calls_pip_sync_and_updates_cache(
+        self, cached_hashes_path, run, requirements_files
+    ):
+        def test_it_calls_pip_sync_and_updates_cache():
+            """Run pip-sync-faster and assert that it calls pip-sync and updates the cached hashes."""
+            args = ["--foo", "--bar", "42"]
+
+            main.pip_sync_maybe(requirements_files.relpaths, args=args)
+
+            run.assert_called_once_with(
+                ["pip-sync", *requirements_files.relpaths, *args], check=True
+            )
+            with open(cached_hashes_path, "r", encoding="utf-8") as handle:
+                cached_hashes = json.load(handle)
+            assert cached_hashes == requirements_files.hashes
+
+        return test_it_calls_pip_sync_and_updates_cache
+
+
+class TestEntryPoint:
+    def test_it(self, pip_sync_maybe):
+        main.entry_point(["requirements/dev.txt", "--foo", "bar"])
+
+        pip_sync_maybe.assert_called_once_with(
+            ["requirements/dev.txt"], ["--foo", "bar"]
+        )
+
+    def test_version(self, capsys):
+        with pytest.raises(SystemExit) as exc_info:
+            main.entry_point(["--version"])
+
+        assert (
+            capsys.readouterr().out.strip()
+            == f"pip-sync-faster, version {version('pip-sync-faster')}"
+        )
+        assert exc_info.value.code is None
+
+    def test_if_pip_sync_fails(self, pip_sync_maybe):
+        pip_sync_maybe.side_effect = CalledProcessError(23, ["pip-sync"])
+
+        with pytest.raises(SystemExit) as exc_info:
+            main.entry_point(["requirements/dev.txt"])
+
+        # It echoes pip-sync's exit code.
+        assert exc_info.value.code == 23
+
+    @pytest.fixture(autouse=True)
+    def pip_sync_maybe(self, mocker):
+        return mocker.patch("pip_sync_faster.main.pip_sync_maybe", autospec=True)
+
+
+class File:
+    """Metadata about a file."""
+
+    def __init__(self, relpath):
+        self.relpath = relpath
+        self.abspath = abspath(relpath)
+
+        hashobj = hashlib.sha512()
+        with open(self.abspath, "rb") as file:
+            hashobj.update(file.read())
+
+        self.hash = hashobj.hexdigest()
```

Comment:
Maybe the tests should just use the `SrcFile` class from `main.py`: it's exactly the same thing as this class.